diff --git a/central_app.py b/central_app.py
new file mode 100644
--- /dev/null
+++ ./central_app.py
@@ -0,0 +1,380 @@
+from PyQt5.QtWidgets import (
+    QDesktopWidget,
+    QApplication,
+    QMainWindow,
+    QScrollArea,
+    QLabel,
+    QShortcut,
+    QFileDialog,
+    QSizePolicy,
+)
+
+from PyQt5.QtCore import (
+    Qt,
+    pyqtSignal,
+    QEvent,
+    QCoreApplication,
+)
+from PyQt5.QtGui import QPixmap, QImage, QKeySequence
+from PyQt5.QtCore import Qt, pyqtSignal, QEvent, QCoreApplication
+from watchdog.observers import Observer
+import sys
+import tempfile
+import threading
+import socket
+import os
+from events import OpenWindowEvent
+from file_change_handler import FileChangeHandler
+from logger import setup_logging
+import logging
+import subprocess
+from PyQt5.QtWidgets import QApplication, QMainWindow
+from PyQt5.QtCore import QTimer
+import os
+import glob
+
+# 使用 pyobjc 导入 AppKit
+from AppKit import NSWorkspace
+
+# 确保日志目录存在
+log_dir = os.path.expanduser("~/loadrc/plantumlmacviewer")
+os.makedirs(log_dir, exist_ok=True)
+
+logging.basicConfig(
+    level=logging.DEBUG,
+    format="%(asctime)s - %(levelname)s - %(message)s",
+    filename=os.path.join(log_dir, "plantumlviewerserver.log"),
+    filemode="a",  # 追加模式
+)
+
+# 立即添加一条日志，确保日志系统正在工作
+logging.info("central_app.py 日志系统初始化完成")
+
+
+class CentralApp(QApplication):
+    def __init__(self, argv):
+        logging.debug("开始初始化 CentralApp")
+        super().__init__(argv)
+        logging.debug("CentralApp 初始化完成")
+        self.windows = []
+        self.fileWindowMap = {}
+        self.observers = {}
+
+        # 启动套接字监听线程
+        logging.debug("准备启动套接字监听线程")
+        socketThread = threading.Thread(target=self.listenToSocket)
+        socketThread.daemon = True
+        socketThread.start()
+        logging.debug("套接字监听线程启动完成")
+
+    def listenToSocket(self):
+        host = "localhost"
+        port = 12345
+
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            s.bind((host, port))
+            s.listen()
+
+            logging.info(f"Listening on {host}:{port}")
+
+            while True:
+                try:
+                    conn, addr = s.accept()
+                    with conn:
+                        logging.info(f"Connected by {addr}")
+                        data = conn.recv(4096).decode().strip()
+                        file_paths = data.split("\n")
+                        logging.info(f"Received file paths: {file_paths}")
+                        for file_path in file_paths:
+                            QCoreApplication.postEvent(
+                                self, OpenWindowEvent([file_path])
+                            )
+                except Exception as e:
+                    logging.error(f"Socket error: {str(e)}")
+
+    def customEvent(self, event):
+        logging.debug(f"customEvent triggered with event type: {event.type()}")
+        if event.type() == OpenWindowEvent.EVENT_TYPE:
+            for filePath in event.filePaths:
+                try:
+                    self.openNewWindow(filePath)
+                except Exception as e:
+                    logging.error(
+                        f"Error opening new window for {filePath}: {str(e)}"
+                    )
+
+    def openNewWindow(self, filePath=None):
+        logging.debug(f"openNewWindow called with filePath: {filePath}")
+        try:
+            # 确保路径是规范化的
+            if filePath and not filePath.startswith("fugitive:///"):
+                filePath = os.path.abspath(filePath)
+
+            # 如果文件已经打开，激活对应的窗口
+            if filePath in self.fileWindowMap:
+                window = self.fileWindowMap[filePath]
+                logging.info(f"Activating existing window for {filePath}")
+                window.raise_()
+                window.activateWindow()
+                QCoreApplication.processEvents()  # 处理事件队列
+                return
+
+            # 创建新窗口并将其添加到窗口列表和文件映射中
+            new_window = UMLViewer(self)
+            self.windows.append(new_window)
+            if filePath:
+                self.fileWindowMap[filePath] = new_window
+                new_window.loadAndDisplayUML(filePath)
+                self.startFileWatcher(filePath, new_window)
+            new_window.show()
+
+            # 激活并将新窗口置于前台
+            new_window.raise_()
+            new_window.activateWindow()
+
+            logging.info(f"New window created and activated for {filePath}")
+        except Exception as e:
+            logging.error(f"Error in openNewWindow: {str(e)}")
+
+    def startFileWatcher(self, filePath, viewer):
+        logging.debug(f"startFileWatcher called with filePath: {filePath}")
+        if not filePath.startswith("fugitive:///"):
+            # 确保路径是规范化的
+            filePath = os.path.abspath(filePath)
+        # 获取目录路径
+        directory = os.path.dirname(filePath)
+
+        # 检查此目录是否已经有一个监控器，如果有，只需添加事件处理器，而不是创建新的监控器
+        if directory in self.observers:
+            # 为已存在的监控器添加事件处理器
+            event_handler = FileChangeHandler(viewer, filePath)
+            self.observers[directory].schedule(
+                event_handler, directory, recursive=False
+            )
+        else:
+            # 创建新的 Observer
+            observer = Observer()
+            self.observers[directory] = observer
+            event_handler = FileChangeHandler(viewer, filePath)
+            observer.schedule(event_handler, directory, recursive=False)
+            observer.start()
+
+
+class UMLViewer(QMainWindow):
+    def __init__(self, centralApp):
+        logging.debug("开始初始化 UMLViewer")
+        super().__init__()
+        self.centralApp = centralApp
+        self.initUI()
+        logging.debug("UMLViewer 初始化完成")
+
+    def initUI(self):
+        logging.debug("开始初始化 UI")
+        self.setWindowTitle("PlantUML Viewer")
+        self.setGeometry(100, 100, 800, 600)
+        logging.debug("UI 初始化完成")
+
+        # 容纳UML图像的滚动区域
+        self.scrollArea = QScrollArea(self)
+        self.setCentralWidget(self.scrollArea)
+
+        # UML图像标签
+        self.imageLabel = QLabel()
+        self.imageLabel.setAlignment(Qt.AlignCenter)  # 居中对齐
+
+        # 设置图像的尺寸策略，使其能够根据可用空间自动调整大小
+        self.imageLabel.setSizePolicy(QSizePolicy.Ignored, QSizePolicy.Ignored)
+        self.imageLabel.setScaledContents(True)  # 允许图像根据标签大小缩放
+        self.scrollArea.setWidget(self.imageLabel)
+        self.scrollArea.setWidgetResizable(True)  # 允许滚动区域适应内容大小
+
+        # 设置快捷键
+        self.setupShortcuts()
+
+        # 将窗口放置在当前显示器上
+        self.move_to_current_screen()
+
+        # 窗口最大化
+        self.showMaximized()
+
+    def move_to_current_screen(self):
+        main_window = (
+            self.centralApp.windows[0] if self.centralApp.windows else None
+        )
+        if main_window:
+            screen = QApplication.desktop().screenNumber(main_window)
+        else:
+            screen = QApplication.desktop().primaryScreen()
+
+        rect = QApplication.desktop().screenGeometry(screen)
+        self.setGeometry(rect)
+
+    def setupShortcuts(self):
+        # 打开文件快捷键
+        self.openFileShortcut = QShortcut(QKeySequence("Ctrl+O"), self)
+        self.openFileShortcut.activated.connect(self.openFile)
+        logging.debug("Open file shortcut (Ctrl+O) set up")
+
+        # 关闭窗口快捷键
+        self.closeWindowShortcut = QShortcut(QKeySequence("Ctrl+W"), self)
+        self.closeWindowShortcut.activated.connect(self.close)
+        logging.debug("Close window shortcut (Ctrl+W) set up")
+
+    def openFile(self):
+        # 修改 openFile 方法以支持在新窗口中打开文件
+        filePath, _ = QFileDialog.getOpenFileName(
+            self, "Open file", "", "PlantUML files (*.puml)"
+        )
+        if not filePath.startswith("fugitive:///"):
+            # 确保路径是规范化的
+            filePath = os.path.abspath(filePath)
+            self.centralApp.openNewWindow(filePath)
+
+    def loadAndDisplayUML(self, filePath):
+        temp_dir = None
+        try:
+            self.previousApp = self.getActiveAppName()
+            self.setFocusPolicy(Qt.NoFocus)
+            # 在加载 UML 之前，设置窗口标题为文件名
+            self.setWindowTitle(os.path.basename(filePath))
+            plantuml_jar_path = self.get_plantuml_jar_path()
+
+            temp_dir = tempfile.mkdtemp()
+            base_name = os.path.basename(filePath)
+            png_name = (
+                base_name.replace(".puml", "").replace(" ", "_") + ".png"
+            )
+            temp_png_path = os.path.join(temp_dir, png_name)
+
+            command = [
+                "java",
+                "-jar",
+                plantuml_jar_path,
+                "-tpng",
+                "-o",
+                temp_dir,
+                filePath,
+            ]
+            logging.info(f"Running command: {' '.join(command)}")
+            result = subprocess.run(command, capture_output=True, text=True)
+            logging.info(f"PlantUML return code: {result.returncode}")
+            logging.info(f"PlantUML Output: {result.stdout}")
+            logging.info(f"PlantUML Error Output: {result.stderr}")
+            logging.info(f"Temp directory contents: {os.listdir(temp_dir)}")
+
+            # 查找生成的PNG文件
+            generated_files = [
+                f for f in os.listdir(temp_dir) if f.endswith(".png")
+            ]
+            if not generated_files:
+                raise FileNotFoundError(f"No PNG file generated in {temp_dir}")
+
+            actual_png_path = os.path.join(temp_dir, generated_files[0])
+            logging.info(f"Found generated PNG file: {actual_png_path}")
+
+            if os.path.getsize(actual_png_path) == 0:
+                raise ValueError(
+                    f"Generated PNG file is empty: {actual_png_path}"
+                )
+
+            self.displayImage(actual_png_path)
+
+        except subprocess.CalledProcessError as e:
+            logging.exception(f"Error during subprocess execution: {e}")
+            self.imageLabel.setText(f"Error generating UML diagram: {e}")
+        except FileNotFoundError as e:
+            logging.exception(f"File not found: {e}")
+            self.imageLabel.setText(f"Error: {e}")
+        except ValueError as e:
+            logging.exception(f"Invalid file: {e}")
+            self.imageLabel.setText(f"Error: {e}")
+        except Exception as e:
+            logging.exception(f"Unexpected error: {e}")
+            self.imageLabel.setText(f"An unexpected error occurred: {e}")
+        finally:
+            self.focusSignal.emit()
+            if temp_dir and os.path.exists(temp_dir):
+                try:
+                    import shutil
+
+                    shutil.rmtree(temp_dir)
+                except Exception as e:
+                    logging.error(f"Error removing temp dir: {e}")
+
+    def get_plantuml_jar_path(self):
+        plantuml_base_path = "/usr/local/Cellar/plantuml/"
+        latest_version_path = max(
+            glob.glob(os.path.join(plantuml_base_path, "*/")),
+            key=os.path.getmtime,
+        )
+        return os.path.join(latest_version_path, "libexec/plantuml.jar")
+
+    def displayImage(self, imagePath):
+        logging.info(f"Loading PNG file: {imagePath}")
+        self.imageLabel.clear()  # 清空现有图像
+        image = QImage(imagePath)
+        if not image.isNull():
+            pixmap = QPixmap.fromImage(image)
+            self.imageLabel.setPixmap(pixmap)
+            logging.info("Image updated successfully.")
+        else:
+            error_msg = f"Failed to load the generated image: {imagePath}"
+            logging.error(error_msg)
+            self.imageLabel.setText(error_msg)
+
+    def postFocusProcessing(self):
+        self.raise_()
+        self.activateWindow()
+        QApplication.processEvents()
+        self.setFocusToApp(self.previousApp)
+
+    def getActiveAppName(self):
+        ws = NSWorkspace.sharedWorkspace()
+        frontmostApp = ws.frontmostApplication()
+        return frontmostApp.localizedName()
+
+    def setFocusToApp(self, appName):
+        ws = NSWorkspace.sharedWorkspace()
+        for app in ws.runningApplications():
+            if app.localizedName() == appName:
+                app.activateWithOptions_(
+                    NSWorkspace.NSApplicationActivateIgnoringOtherApps
+                )
+                break
+
+    def closeEvent(self, event):
+        logging.debug("Close event triggered")
+        # Remove this window from the centralApp's windows list
+        if self in self.centralApp.windows:
+            self.centralApp.windows.remove(self)
+
+        # Remove this window from the fileWindowMap
+        for file_path, window in list(self.centralApp.fileWindowMap.items()):
+            if window == self:
+                del self.centralApp.fileWindowMap[file_path]
+
+        # Accept the close event
+        event.accept()
+
+
+def main():
+    logging.info("central_app.py 开始执行")
+    try:
+        app = CentralApp(sys.argv)
+        logging.info("CentralApp 创建完成")
+
+        viewer = UMLViewer(app)
+        viewer.show()
+        logging.info("UMLViewer 显示完成")
+
+        # 添加一个定时器，在应用程序开始运行后打印一条消息
+        QTimer.singleShot(100, lambda: logging.info("应用程序事件循环已开始运行"))
+
+        sys.exit(app.exec_())
+    except Exception as e:
+        logging.exception(f"发生异常: {str(e)}")
+        sys.exit(1)
+
+
+if __name__ == "__main__":
+    main()
diff --git a/events.py b/events.py
new file mode 100644
--- /dev/null
+++ ./events.py
@@ -0,0 +1,9 @@
+from PyQt5.QtCore import QEvent
+
+
+class OpenWindowEvent(QEvent):
+    EVENT_TYPE = QEvent.Type(QEvent.registerEventType())
+
+    def __init__(self, filePaths):
+        super().__init__(OpenWindowEvent.EVENT_TYPE)
+        self.filePaths = filePaths  # filePaths 现在是一个列表
diff --git a/file_change_handler.py b/file_change_handler.py
new file mode 100644
--- /dev/null
+++ ./file_change_handler.py
@@ -0,0 +1,12 @@
+from watchdog.events import FileSystemEventHandler
+
+
+class FileChangeHandler(FileSystemEventHandler):
+    def __init__(self, viewer, filePath):
+        super().__init__()
+        self.viewer = viewer
+        self.filePath = filePath
+
+    def on_modified(self, event):
+        if event.src_path == self.filePath:
+            self.viewer.loadAndDisplayUML(self.filePath)
diff --git a/logger.py b/logger.py
new file mode 100644
--- /dev/null
+++ ./logger.py
@@ -0,0 +1,17 @@
+import logging
+import os
+
+
+def setup_logging():
+    # 获取当前文件的绝对路径
+    current_file_path = os.path.abspath(__file__)
+    # 获取当前文件所在的目录
+    current_dir = os.path.dirname(current_file_path)
+    # 创建日志文件路径，位于当前目录下
+    log_file = os.path.join(current_dir, "plantumlviewerserver.log")
+
+    logging.basicConfig(
+        filename=log_file,
+        level=logging.DEBUG,  # 将日志级别设置为DEBUG以捕获所有日志
+        format="%(asctime)s %(levelname)s:%(message)s",
+    )
diff --git a/main.py b/main.py
new file mode 100644
--- /dev/null
+++ ./main.py
@@ -0,0 +1,109 @@
+import os
+import sys
+import venv
+import time
+import subprocess
+from central_app import CentralApp
+import logging
+
+# 确保日志目录存在
+log_dir = os.path.expanduser("~/loadrc/plantumlmacviewer")
+os.makedirs(log_dir, exist_ok=True)
+
+# 配置日志
+logging.basicConfig(
+    level=logging.DEBUG,
+    format="%(asctime)s - %(levelname)s - %(message)s",
+    filename=os.path.join(log_dir, "plantumlviewerserver.log"),
+    filemode="a",  # 追加模式
+)
+
+
+# 立即添加一条日志，确保日志系统正在工作
+logging.info("日志系统初始化完成")
+
+
+def create_or_get_venv(venv_name):
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    venv_path = os.path.join(current_dir, venv_name)
+
+    if not os.path.exists(venv_path):
+        logging.info(f"正在创建虚拟环境: {venv_name}")
+        venv.create(venv_path, with_pip=True)
+        return venv_path, True  # 返回路径和一个标志，表示是新创建的
+    else:
+        logging.info(f"虚拟环境 {venv_name} 已存在")
+        return venv_path, False  # 返回路径和一个标志，表示不是新创建的
+
+
+def get_venv_python(venv_path):
+    if sys.platform == "win32":
+        return os.path.join(venv_path, "Scripts", "python.exe")
+    else:
+        return os.path.join(venv_path, "bin", "python")
+
+
+def install_requirements(venv_python):
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    requirements_path = os.path.join(current_dir, "requirements.txt")
+    if os.path.exists(requirements_path):
+        logging.info("正在安装依赖包...")
+        subprocess.check_call(
+            [venv_python, "-m", "pip", "install", "-r", requirements_path]
+        )
+    else:
+        logging.warning("未找到requirements.txt文件。跳过包安装。")
+
+
+def main():
+    logging.info("开始执行main函数")
+    venv_name = "plantuml_viewer_env"
+    venv_path, is_new_venv = create_or_get_venv(venv_name)
+    venv_python = get_venv_python(venv_path)
+
+    if is_new_venv:
+        install_requirements(venv_python)
+    else:
+        logging.info("使用现有虚拟环境，跳过依赖安装")
+
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    central_app_path = os.path.join(current_dir, "central_app.py")
+
+    logging.info(f"准备运行central_app.py，路径：{central_app_path}")
+
+    env = os.environ.copy()
+    if sys.platform == "win32":
+        site_packages = os.path.join(venv_path, "Lib", "site-packages")
+    else:
+        site_packages = os.path.join(
+            venv_path,
+            "lib",
+            f"python{sys.version_info.major}.{sys.version_info.minor}",
+            "site-packages",
+        )
+    env["PYTHONPATH"] = f"{site_packages}:{env.get('PYTHONPATH', '')}"
+
+    process = subprocess.Popen(
+        [venv_python, central_app_path],
+        env=env,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        text=True,
+    )
+
+    # 等待一段时间，确保服务器有足够时间启动
+    time.sleep(5)
+
+    while True:
+        output = process.stdout.readline()
+        if output == "" and process.poll() is not None:
+            break
+        if output:
+            print(output.strip())
+
+    rc = process.poll()
+    logging.info(f"central_app.py 执行完成，返回码：{rc}")
+
+
+if __name__ == "__main__":
+    main()
diff --git a/requirements.txt b/requirements.txt
new file mode 100644
--- /dev/null
+++ ./requirements.txt
@@ -0,0 +1,3 @@
+PyQt5
+watchdog
+pyobjc
